cpg_workflows/large_cohort/scripts/browser_vcf_release.py

```diff
@@ -0,0 +1,2396 @@
+                    'AS_ReadPosRankSum',
+                    'AS_FS',
+                    'AS_SOR',
+                ],
+            ),
+        )
+        inbreeding_coeff_cutoff = hl.float64(
+            config_retrieve(['large_cohort', 'browser', 'inbreeding_coeff_cutoff']),
+        )
+        ht = ht.annotate_globals(
+            inbreeding_coeff_cutoff=inbreeding_coeff_cutoff,
+        )
+
+    if not for_joint:
+        # v4 Genomes drops subsets from VCF
+        subsets = SUBSETS['exomes'] if data_type == 'exomes' else []
+        header_dict = prepare_vcf_header_dict(
+            ht,
+            validated_ht=validated_ht,
+            info_fields=list(validated_ht.info),
+            bin_edges=make_hist_bin_edges_expr(
+                ht,
+                data_type=data_type,
+                include_age_hists=True,
+            ),
+            age_hist_distribution=hl.eval(ht.age_distribution.bin_freq),
+            subset_list=subsets,
+            pops=GEN_ANC_GROUPS[data_type],  # type: ignore[arg-type]
+            data_type=data_type,
+            joint_included=joint_included,
+        )
+    else:
+        header_dict = {'filter': make_vcf_filter_dict(joint=True), 'info': {}}  # type: ignore[dict-item]
+        for dt in ['exomes', 'genomes', 'joint']:
+            dt_ht = select_type_from_joint_ht(ht, dt)
+            temp_header_dict = prepare_vcf_header_dict(
+                dt_ht,
+                validated_ht=validated_ht,
+                info_fields=[f for f in validated_ht.info.keys() if dt in f],
+                bin_edges=make_hist_bin_edges_expr(
+                    dt_ht,
+                    data_type=data_type,
+                    include_age_hists=True,
+                    for_joint=for_joint,
+                ),
+                age_hist_distribution=hl.eval(dt_ht.age_distribution.bin_freq),
+                subset_list=[dt],
+                pops=GEN_ANC_GROUPS[dt],  # type: ignore[arg-type]
+                data_type='joint',
+                joint_included=joint_included,
+                freq_comparison_included=(dt == 'joint'),
+                extra_suffix=dt,
+                extra_description_text=f' in {dt} dataset',
+            )
+            header_dict['info'].update(temp_header_dict)  # type: ignore[arg-type]
+
+    with hl.hadoop_open(output_path(f'{contig}_{data_type}_header_dict.pkl', category='tmp'), 'wb') as p:
+        pickle.dump(header_dict, p, protocol=pickle.HIGHEST_PROTOCOL)
+
+    ht = validated_ht
+
+    if contig:
+        ht = hl.filter_intervals(
+            ht,
+            [hl.parse_locus_interval(contig, reference_genome='GRCh38')],
+        )
+
+    ht, new_row_annots = format_validated_ht_for_export(ht, data_type=data_type)
+
+    ordered_vcf_info_dict = {f: header_dict['info'][f] for f in list(ht.info) if f in header_dict['info']}
+    header_dict.update({'info': ordered_vcf_info_dict})
+
+    hl.export_vcf(
+        ht,
+        f'{contig}_{vcf_outpath}',
```

Comment:
This method allows `contig` to be None, so this output path should probably account for that